src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -49,52 +50,45 @@ public class UpdateSecretsManagerSubscriptionCommand : IUpdateSecretsManagerSubs
         _referenceEventService = referenceEventService;
     }
 
-    public async Task UpdateSecretsManagerSubscription(OrganizationUpdate update)
+    public async Task UpdateSecretsManagerSubscription(SecretsManagerSubscriptionUpdate update)
     {
         var organization = await _organizationRepository.GetByIdAsync(update.OrganizationId);
 
         if (organization == null)
         {
-            throw new NotFoundException();
+            throw new NotFoundException("Organization is not found");
         }
 
-        var newSeatCount = organization.SmSeats.HasValue
-            ? organization.SmSeats.Value + update.SeatAdjustment
-            : update.SeatAdjustment;
-
-
-        if (update.MaxAutoscaleSeats.HasValue && newSeatCount > update.MaxAutoscaleSeats.Value)
+        if (!organization.UseSecretsManager)
         {
-            throw new BadRequestException("Cannot set max seat autoscaling below seat count.");
+            throw new BadRequestException("Organization has no access to Secrets Manager.");
         }
 
-        var newServiceAccountCount = organization.SmServiceAccounts.HasValue
-            ? organization.SmServiceAccounts + update.ServiceAccountsAdjustment
-            : update.ServiceAccountsAdjustment;
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
 
-        if (update.MaxAutoscaleServiceAccounts.HasValue && newServiceAccountCount > update.MaxAutoscaleServiceAccounts.Value)
+        if (plan == null)
         {
-            throw new BadRequestException("Cannot set max service account autoscaling below service account count.");
+            throw new BadRequestException("Existing plan not found.");
         }
 
         if (update.SeatAdjustment != 0)
         {
-            await AdjustSeatsAsync(organization, update.SeatAdjustment);
+            await AdjustSeatsAsync(organization, update, plan);
         }
 
         if (update.ServiceAccountsAdjustment != 0)
         {
-            await AdjustServiceAccountsAsync(organization, update.ServiceAccountsAdjustment.GetValueOrDefault());
+            await AdjustServiceAccountsAsync(organization, update, plan);
         }
 
         if (update.MaxAutoscaleSeats != organization.MaxAutoscaleSmSeats)
```

Comment:
I am not sure we need to check that organization.MaxAutoscaleSeats.HasValue because it might be wrong for the first time and we can still autoscale the seats.